File: Largest subtree sum in a tree.cpp
class Solution {
  public:
    int findLargestSubtreeSumUtil(Node* root, int& ans)
{
    // If current node is null then
    // return 0 to parent node.
    if (root == NULL)    
        return 0;
     
    // Subtree sum rooted at current node.
    int currSum = root->data +
      findLargestSubtreeSumUtil(root->left, ans)
      + findLargestSubtreeSumUtil(root->right, ans);
 
    // Update answer if current subtree
    // sum is greater than answer so far.
    ans = max(ans, currSum);
 
    // Return current subtree sum to
    // its parent node.
    return currSum;
}
 
// Function to find largest subtree sum.
int findLargestSubtreeSum(Node* root)
{
    // If tree does not exist,
    // then answer is 0.
    if (root == NULL)    
        return 0;
     
    // Variable to store maximum subtree sum.
    int ans = INT_MIN;
 
    // Call to recursive function to
    // find maximum subtree sum.
    findLargestSubtreeSumUtil(root, ans);
 
    return ans;
}
};

File: Largest subtree sum in a tree.py
class Solution:
    def findLargestSubtreeSumUtil(self,root):
        if root==None:
            return 0
        else:
            currsum=root.data+self.findLargestSubtreeSumUtil(root.left)+self.findLargestSubtreeSumUtil(root.right)
            self.ans=max(self.ans,currsum)
            return currsum
        
    def findLargestSubtreeSum(self, root ) -> int:
        self.ans=-float("inf")
        if root==None:
            return 0
        self.findLargestSubtreeSumUtil(root)
        return self.ans
